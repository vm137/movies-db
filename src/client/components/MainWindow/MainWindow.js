import React, {Component} from 'react';
import SearchBlock from '../SearchBlock/SearchBlock';
import SingleMovie from '../SingleMovie/SingleMovie';

import './MainWindow.scss';
import fakeData from "../constants/fakeData";

export default class MainWindow extends Component {

    constructor(props) {
        super(props);
        this.state = {
            searchBlock: true
        };
    }

    componentWillMount() {
        this.makeFetch();
    }

    makeFetch() {
        /* eslint-disable no-undef */
        fetch("http://react-cdp-api.herokuapp.com/movies")
            .then(function (response) {
                return response.json();
            })
            .then(myJson => {
                    // console.log(myJson);
                    this.state.movies = myJson;
                    console.log(this.state.movies);
                }
            )
        //     .catch(
        //     (error) => {
        //         console.log("=== fetch unsuccessful.");
        //         console.log(error);
        //
        //         this.state.movies = fakeData;
        //     }
        // );
    }

    render() {
        return (
          <div>
            { this.state.searchBlock ? <SearchBlock /> : <SingleMovie /> }
          </div>
        );
    }
}
