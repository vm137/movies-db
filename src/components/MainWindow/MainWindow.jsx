// @flow

import React from 'react';
import { Route, Switch, Redirect } from 'react-router-dom';
import SearchBlock from '../SearchBlock';
import SingleMovie from '../SingleMovie';
import NoMatch from '../NoMatch';
import './style.scss';

type Props = {
  Router: Function,
  location: string,
  context: Object
}

const MainWindow = ({ Router, location, context }: Props) => (
  <div className="mainWindow">
    <Router location={location} context={context}>
      <Switch>
        <Route exact path="/" render={() => <Redirect to="/search" />} />
        <Route path="/search/:query?" component={SearchBlock} />
        <Route path="/film/:id" component={SingleMovie} />
        <Route component={NoMatch} />
      </Switch>
    </Router>
  </div>
);

export default MainWindow;
