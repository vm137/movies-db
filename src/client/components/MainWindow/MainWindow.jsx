import React, {Component} from 'react';
import SearchBlock from '../SearchBlock/SearchBlock';
import SingleMovie from '../SingleMovie/SingleMovie';
import './MainWindow.scss';


export default class MainWindow extends Component {

    constructor(props) {
        super(props);
        this.makeFetch = this.makeFetch.bind(this);
        this.state = {
            searchBlock: true,
            foundMovies: 0,
            movies: []
        };
    }

    makeFetch(searchString = '', searchBy = '', offset = '', limit = '') {
        let callString = 'http://react-cdp-api.herokuapp.com/movies';
        callString += `?search=${searchString}&searchBy=${searchBy}&offset=${offset}&limit=${limit}`;

        /* eslint-disable no-undef */
        fetch(callString)
            .then(function (response) {
                return response.json();
            })
            .then(myJson => {
                this.setState({
                    movies: myJson.data,
                    foundMovies: myJson.total
                })
            }
            )
        //     .catch(
        //     (error) => {
        //         console.log("=== fetch unsuccessful.");
        //         console.log(error);
        //
        //         this.state.movies = fakeData;
        //     }
        // );
    }

    render() {
        return (
          <div>
            { this.state.searchBlock ?
              <SearchBlock movies={this.state.movies} foundMovies={this.state.foundMovies} searchCB={this.makeFetch} /> :
              <SingleMovie movie="single movie 1" movies={this.state.movies} />
            }
          </div>
        );
    }
}
