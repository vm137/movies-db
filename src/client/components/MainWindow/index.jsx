import { connect } from 'react-redux';
import MainWindow from './MainWindow';

const mapStateToProps = state => ({
  showPage: state.showPage,
});

export default connect(mapStateToProps)(MainWindow);
