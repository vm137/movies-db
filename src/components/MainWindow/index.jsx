import { connect } from 'react-redux';
import MainWindow from './MainWindow';

const mapStateToProps = ({ showPage }) => ({ showPage });

export default connect(mapStateToProps)(MainWindow);
