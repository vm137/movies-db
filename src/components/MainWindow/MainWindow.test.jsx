import { shallow } from 'enzyme';
import React from 'react';
import MainWindow from './MainWindow';

describe('<MainWindow />', () => {
  it('renders matching snapshot', () => {
    const wrapper = shallow(<MainWindow />);
    expect(wrapper).toMatchSnapshot();
  });
});
