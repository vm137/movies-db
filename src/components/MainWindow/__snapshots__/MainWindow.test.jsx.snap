// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainWindow /> renders matching snapshot 1`] = `
<div
  className="mainWindow"
>
  <Component>
    <Switch>
      <Route
        exact={true}
        path="/"
        render={[Function]}
      />
      <Route
        component={[Function]}
        path="/search/:query?"
      />
      <Route
        component={[Function]}
        path="/film/:id"
      />
      <Route
        component={[Function]}
      />
    </Switch>
  </Component>
</div>
`;
